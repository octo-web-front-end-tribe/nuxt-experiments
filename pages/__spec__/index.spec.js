import {shallow} from '@vue/test-utils';
import Logo from '~/components/Logo';

import Home from '../index.vue';

describe('root page', () => {
  it('should match snapshot', () => {
    // when
    const wrapper = shallow(Home);

    // then
    expect(wrapper.element).toBeDefined();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should contain logo component', () => {
    // given
    const wrapper = shallow(Home);

    // then
    let logoWrappers = wrapper.findAll(Logo);
    expect(logoWrappers.length).toEqual(1);
  });

  describe('title', () => {
    it('should contain default message', () => {
      // when
      const wrapper = shallow(Home);

      // then
      const title = wrapper.find('h1');
      expect(title.text()).toContain('Welcome'); // default
    });

    it('should update when msg data is changed', () => {
      // when
      const wrapper = shallow(Home);

      // then
      const title = wrapper.find('h1');
      wrapper.setData({msg: 'test'});
      expect(title.text()).toContain('test');
    });
  });
});
