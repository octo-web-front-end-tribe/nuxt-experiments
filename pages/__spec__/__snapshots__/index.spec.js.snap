// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`root page should match snapshot 1`] = `
<section
  class="container"
>
  <!---->
   
  <h1
    class="title"
  >
    
    Welcome
  
  </h1>
   
  <div>
    <nuxt-link
      to="/counter"
    >
      Counter
    </nuxt-link>
     
    <nuxt-link
      exact=""
      to="/everyone"
    >
      Welcome who ?
    </nuxt-link>
  </div>
   
  <div
    class="links"
  >
    <a
      class="button--green"
      href="https://nuxtjs.org/"
      target="_blank"
    >
      Documentation
    </a>
     
    <a
      class="button--grey"
      href="https://github.com/nuxt/nuxt.js"
      target="_blank"
    >
      GitHub
    </a>
  </div>
</section>
`;
